jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/HttpOutput.java

```diff
@@ -576,25 +577,32 @@ public ByteBuffer getByteBuffer()
     {
         try (AutoLock l = _channelState.lock())
         {
-            return acquireBuffer().getByteBuffer();
+            return lockedAcquireBuffer().getByteBuffer();
         }
     }
 
-    private RetainableByteBuffer acquireBuffer()
+    private RetainableByteBuffer lockedAcquireBuffer()
     {
         boolean useOutputDirectByteBuffers = _servletChannel.getConnectionMetaData().getHttpConfiguration().isUseOutputDirectByteBuffers();
-        ByteBufferPool pool = _servletChannel.getRequest().getComponents().getByteBufferPool();
+
         if (_aggregate == null)
-            _aggregate = pool.acquire(getBufferSize(), useOutputDirectByteBuffers);
+        {
+            _pool = _servletChannel.getRequest().getComponents().getByteBufferPool();
+            _aggregate = _pool.acquire(getBufferSize(), useOutputDirectByteBuffers);
+        }
         return _aggregate;
     }
 
-    private void releaseBuffer()
+    private void lockedReleaseBuffer(boolean failure)
```

Comment:
If you prefix this method's name with `locked` because a lock needs to be held before it is called, you should add as its 1st line an assertion that checks the lock is held.